nncf/common/quantization/quantizer_propagation/graph.py

```diff
@@ -1422,6 +1422,27 @@ def _handle_output_quantizers_for_weights_as_outputs_ops(
             setup.discard(qp_id_for_current_pq, keep_shared_input_qps=True)
         return setup
 
+    @staticmethod
+    def _get_weight_and_activation_qconfig_list_intersection(
+        weight_qconfig_options: List[QuantizerConfig], activation_qconfig_options: List[QuantizerConfig]
+    ) -> List[QuantizerConfig]:
+        """
+        Returns special intersection between weight and activation quantization configurations.
+
+        :param weight_qconfig_options: List of QuantizerConfig associated with weights.
+        :param activation_qconfig_options: List of QuantizerConfig associated with activations.
+        :return: Special intersection between configurations.
+        """
+        act_qconfig_extend_list = []
+        for act_qconfig in activation_qconfig_options:
+            if act_qconfig.signedness_to_force is None:
+                for signedness_to_force_position in [True, False]:
+                    act_qconfig_updated = deepcopy(act_qconfig)
+                    act_qconfig_updated.signedness_to_force = signedness_to_force_position
+                    act_qconfig_extend_list.append(act_qconfig_updated)
+        act_qconfig_extend_list += activation_qconfig_options
+        return [qconf for qconf in weight_qconfig_options if qconf in act_qconfig_extend_list]
```

Comment:
Make sure that this logic covers the following case:
let `weight_qconfig_options = [(*partial_configuration*, signedness_to_force=None), ]`

let `activation_qconfig_options = [(*same_partial_configuration*, signedness_to_force=None), ]`

then the intersection should be:
`intersection == [(*same_partial_configuration*, signedness_to_force=None),]`

and currently it looks like your logic will produce the following instead:
`intersection == [(*same_partial_configuration*, signedness_to_force=True), (*same_partial_configuration*, signedness_to_force=False)]`

While we are not likely to encounter this case realistically, I think that the current behaviour in this case would be confusing.